### ISSUE TITLE
Assertion `result <= m_scratchSpaceSizeLimit' on neo runtime debug build

### ISSUE THREAD

**Issue by againull** [intel/intel-graphics-compiler#183]
NEO runtime debug build fails with the following compile-time assertion in IGC:
ocloc: /export/iusers/againull/graphics_compiler/compiler/igc/IGC/Compiler/CodeGenPublic.h:164: unsigned int IGC::SProgramOutput::getScratchSpaceUsageInSlot0() const: Assertion `result <= m_scratchSpaceSizeLimit' failed.
ninja: build stopped: subcommand failed.

This looks like a regression after this commit:
https://github.com/intel/intel-graphics-compiler/commit/606c7ad8709a23f3ac60d34442e38948502b71fc#diff-d08faddc680422a2a86c42d8d213309e1bd1c08eaf6b705d97790b2cd5134681R166

Steps to reproduce - build neo runtime with debug build type using debug version of IGC:
cmake -GNinja -DCMAKE_BUILD_TYPE=Debug -DSKIP_UNIT_TESTS=1 -DIGC_DIR=[path to IGC] -DGMM_DIR=[path to GMM] ..

**Comment by againull** [comment]
I've built Debug version of IGC and GMM per instructions in https://github.com/intel/intel-graphics-compiler and https://github.com/intel/gmmlib.
Then I used the following commands to build ocloc and got the error:
```
git clone https://github.com/intel/compute-runtime.git
cd compute-runtime
mkdir build
cd build
cmake -GNinja -DLevelZero_INCLUDE_DIR=/rdrive/ref/opencl/runtime/linux/level_zero_loader/1.2.3/include -DBUILD_WITH_L0=1 -DCMAKE_BUILD_TYPE=Debug -DSKIP_UNIT_TESTS=1 -DIGC_DIR=/export/iusers/againull/graphics_compiler/compiler/install/usr/local -DGMM_DIR=/export/iusers/againull/graphics_compiler/gmmlib/install/usr/local ..
ninja ocloc
```

Note. Level Zero header files can be cloned from here if needed: https://github.com/oneapi-src/level-zero/tree/master/include

**Comment by againull** [comment]
It looks like problem is fixed somewhere between 4a48ab06741e6cc764bf325917efc4f9dc6de3da and d8d71a22a2595100728d0f3fe05a0486d19f4da9.
